system_metrics_collector/src/system_metrics_collector/linux_memory_measurement_node.cpp

```diff
@@ -124,4 +125,30 @@ double LinuxMemoryMeasurementNode::periodicMeasurement()
   return processMemInfoLines(read_string);
 }
 
+void LinuxMemoryMeasurementNode::publishStatistics()
+{
+  if (publisher_ == nullptr) {
+    return;
+  }
+
+  const moving_average_statistics::StatisticData statistic_data = getStatisticsResults();
```

Comment:
This block is duplicated: let's make a helper method in utility functions so we can reuse.

Reply:
After some refactoring, this block is no longer duplicated.